plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineSource.java

```diff
@@ -0,0 +1,226 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.docker.machine;
+
+import com.google.common.base.MoreObjects;
+
+import org.eclipse.che.api.core.model.machine.MachineSource;
+import org.eclipse.che.api.machine.server.model.impl.MachineSourceImpl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.eclipse.che.plugin.docker.machine.DockerInstanceProvider.DOCKER_IMAGE_TYPE;
+
+/**
+ * Set of helper methods that identifies docker image properties
+ *
+ * @author Sergii Kabashnyuk
+ * @author Florent Benoit
+ */
+public class DockerMachineSource extends MachineSourceImpl {
+
+    /**
+     * Regexp used to parse a location of Docker image.
+     * it's based on optional registry followed by repository name followed by a digest or a tag
+     * Examples are available in the test.
+     */
+    public static final Pattern IMAGE_PATTERN = Pattern.compile("^((?<registry>[^/]++)(?:\\/))?(?<repository>.*?)((?:\\:)(?<tag>.*?))?((?:@)(?<digest>.*))?$");
```

Comment:
As far as I can see this regexp is not tested. Please review your tests.
Also it looks like 
**user/repo**  will be treated as registry **user** and repository **repo**